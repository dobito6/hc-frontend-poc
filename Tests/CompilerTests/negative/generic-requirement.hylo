trait P {
  fun f<T>(_ x: T)
}

given Void: P {
  // This one is OK.
  public fun f<U>(_ x: U) {}
}

struct S<T> {}
given<T> S<T>: P {
  // This one isn't.
  public fun f(_ x: T) {}
}
