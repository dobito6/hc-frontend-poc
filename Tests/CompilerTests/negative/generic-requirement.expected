negative/generic-requirement.hylo:11.1-6: error: no implementation of 'f(_:)'
given<T> S<T>: P {
~~~~~
